server/app/controllers/admin/AdminProgramBlocksController.java

```diff
@@ -44,28 +46,59 @@ public final class AdminProgramBlocksController extends CiviFormController {
   public AdminProgramBlocksController(
       ProgramService programService,
       QuestionService questionService,
-      ProgramBlockEditView.Factory editViewFactory,
+      ProgramBlockEditView.Factory programBlockViewFactory,
       FormFactory formFactory,
       RequestChecker requestChecker) {
     this.programService = checkNotNull(programService);
     this.questionService = checkNotNull(questionService);
-    this.editView = checkNotNull(editViewFactory.create(DRAFT));
+    this.editView = checkNotNull(programBlockViewFactory.create(DRAFT));
+    this.readOnlyView = checkNotNull(programBlockViewFactory.create(ACTIVE));
     this.formFactory = checkNotNull(formFactory);
     this.requestChecker = checkNotNull(requestChecker);
   }
 
   /**
-   * Return a HTML page displaying all configurations of the specified program.
+   * Returns an HTML page displaying all configurations of the specified program and UI elements to
+   * start editing aspects of the program.
```

Comment:
**nitpick:** I'm not sure if this javadoc is correct. I think this page shows the edit page for the last block of the program?

Reply:
Thank you for the clear explanation!